packages/exchange/src/pods/account/account.controller.ts

```diff
@@ -12,30 +13,58 @@ import {
     ClassSerializerInterceptor,
     ValidationPipe,
     UsePipes,
+    BadRequestException,
+    Post,
+    Logger,
+    InternalServerErrorException,
     HttpStatus
 } from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
 
 import { ApiBearerAuth, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { AccountService } from './account.service';
 import { AccountDTO } from './account.dto';
+import { AccountAlreadyExistsError } from './account-already-exists.error';
 
 @ApiTags('account')
 @ApiBearerAuth('access-token')
 @Controller('account')
 @UseInterceptors(NullOrUndefinedResultInterceptor)
+@UseInterceptors(ClassSerializerInterceptor)
 @UsePipes(ValidationPipe)
 export class AccountController {
+    private readonly logger = new Logger(AccountService.name);
+
     constructor(private readonly accountService: AccountService) {}
 
-    // TODO: explicit account creation request
-    @UseInterceptors(ClassSerializerInterceptor)
     @Get()
-    @UseGuards(AuthGuard(), ActiveUserGuard)
+    @UseGuards(AuthGuard())
     @ApiResponse({ status: HttpStatus.OK, type: AccountDTO, description: 'Get the Account' })
-    public async getAccount(@UserDecorator() user: ILoggedInUser): Promise<AccountDTO> {
-        const account = await this.accountService.getAccount(user.ownerId.toString());
+    public async getAccount(@UserDecorator() { ownerId }: ILoggedInUser): Promise<AccountDTO> {
+        const account = await this.accountService.getAccount(ownerId);
+
+        const response = account || { address: '' };
+
+        return response;
+    }
+
+    @Post()
+    @UseGuards(AuthGuard(), ActiveUserGuard, ActiveOrganizationGuard)
+    @ApiResponse({ status: HttpStatus.ACCEPTED, description: 'Created the Exchange Account' })
+    public async create(@UserDecorator() { ownerId }: ILoggedInUser): Promise<boolean> {
+        try {
+            await this.accountService.create(ownerId);
+            return true;
+        } catch (error) {
+            this.logger.error(error.message);
+
+            if (error instanceof AccountAlreadyExistsError) {
+                throw new BadRequestException({ message: error.message });
+            }
 
-        return account;
+            throw new InternalServerErrorException({
+                message: `Unable to create an account due an unknown error`
```

Comment:
Maybe add `Unable to create an account due an unknown error ${error.message}` to the message description so that the user can see the actual error message.

Reply:
This is internal error so nothing usefull for the user.

Reply:
You're right. We need to make sure this is logged somewhere on the server then